src/time-lru-set.js

```diff
@@ -0,0 +1,164 @@
+//
+// Copyright 2019 DxOS.
+//
+
+class ItemCache {
+  constructor ({ value, maxAge, onTimeout }) {
+    this._value = value;
+    this._maxAge = maxAge;
+    this._onTimeout = value => onTimeout(value);
+  }
+
+  update () {
+    this.clear();
+
+    this._timer = setTimeout(() => {
+      this._onTimeout(this._value);
+    }, this._maxAge);
+  }
+
+  clear () {
+    if (this._timer) {
+      clearTimeout(this._timer);
+      this._timer = null;
+    }
+  }
+}
+
+/**
+ * Least Recently Used cache "set" structure with time live support.
+ *
+ */
+class TimeLRUSet {
+  /**
+   * @param {Object} options
+   * @param {number} [options.maxAge=10000] Maximum live time for a value not updated recently.
+   * @param {number} [options.maxSize=100] Max size of the cache.
+   */
+  constructor (options = {}) {
+    const { maxAge = 10 * 1000, maxSize = 100 } = options;
+
+    this._maxAge = maxAge;
+    this._maxSize = maxSize;
+
+    this._list = [];
+    this._map = new Map();
```

Comment:
comment or rename
Map of value timers indexed by value.

Reply:
fixed